server/server.js

```diff
@@ -1,29 +1,160 @@
+  c.loginCallback = function(req, done) {
+    return async function(err, user, identity, token) {
+      var authInfo = {
+        identity: identity,
+      };
+      if (token) {
+        authInfo.accessToken = token;
+      }
+      const profile = identity.profile, provider = identity.provider;
+      const profileUrl = provider == 'github' || provider == 'facebook' ? profile.profileUrl :
+                         (provider == 'twitter' ? "https://twitter.com/" + profile.username : "");
+      const name = provider == 'github' || provider == 'twitter' ? profile.displayName :
+                   provider == 'facebook' ? profile._json.first_name + " " + profile._json.last_name : " ";
+      var account = await app.models.Account.findOne({ where: { ethereumAccountAddress: req.session.ethereumAccountAddress }});
+      if (!account) {
+        account = new app.models.Account();
+        account.ethereumAccountAddress = req.session.ethereumAccountAddress;
+        account.name = name;
+      }
+      account.userId = user.id;
+      account[provider + "URL"] = profileUrl;
+      account.save({ skipSignatureCheck: true });
+
+      // Tell the client we are done so it can close the popup window
+      const io = req.app.get('io');
+      io.in(req.session.socketId).emit(provider, account);
+
+      done(err, user, authInfo);
+    };
+  };
+  passportConfigurator.configureProvider(s, c);
+}
+
+// enable cors
+var corsOption = {
+  origin: true,
+  methods: 'GET,HEAD,PUT,PATCH,POST,DELETE,OPTIONS',
+  credentials: true,
+  exposedHeaders: ['x-auth-token']
+};
+app.use(cors(corsOption));
+
+app.middleware('session:after', function addAccountAddressToSession(req, res, next) {
+  if (req.query.ethereumAccountAddress) {
+    req.session.ethereumAccountAddress = req.query.ethereumAccountAddress;
+  }
+  next();
+});
+
+app.middleware('session:after', function addSocketIdtoSession(req, res, next) {
+  if (req.query.socketId) {
+    req.session.socketId = req.query.socketId;
+  }
+  next();
+});
+
+// Check for local SSL certificate
+let privateKey, certificate, useHTTPS = false;
+try {
+  privateKey = fs.readFileSync(path.join(__dirname, './private/privatekey.pem')).toString();
+  certificate = fs.readFileSync(path.join(__dirname, './private/certificate.pem')).toString();
+  useHTTPS = true;
+} catch(e) {}
+
 app.start = function() {
-  // start the web server
-  return app.listen(function() {
-    app.emit('started');
-    var baseUrl = app.get('url').replace(/\/$/, '');
-    console.log('Web server listening at: %s', baseUrl);
+  var server = null;
+  if (useHTTPS) {
+    var options = {
+      key: privateKey,
+      cert: certificate
+    };
+    server = https.createServer(options, app);
+  } else {
+    server = http.createServer(app);
+  }
+  server.listen(app.get('port'), function() {
+    var baseUrl = (useHTTPS ? 'https://' : 'http://') + app.get('host') + ':' + app.get('port');
```

Comment:
This allowing of non-ssl connections is only for testing pruposes, I presume?

Reply:
yes